### ISSUE TITLE
SEIR Module Extension -- Fix Errors

### ISSUE THREAD

**Issue by smjenness** [statnet/EpiModel#424]
@andsv2 : the SEIRModuleExtension.R does not run without error. 

Here is the error I get after running netsim on line 102:

```r
> sim <- netsim(est, param, init, control)

Starting Network Simulation...
 Error in do.call(control[["prevalence.FUN"]], list(dat, at = 1)) : 
  'what' must be a function or character string 
```

It appears that prevalence.net is not getting set as the default function to prevalence.FUN. Why not?

**Comment by smjenness** [comment]
The `SIRwithVitalDynamics.R` file also does not run without error. It is missing an `r.num` initial condition.

**Comment by smjenness** [comment]
Also in the worked examples in the Rmd and accompanying R script, the SI model with no vital dynamics has the `resimulate.network` control setting set to TRUE (why would network resimulation be needed?). And the SIR model *with* vital dynamics has the `resimulate.network` setting set to FALSE (whereas here it would be needed!). Could you explain the thinking here, and correct as necessary.

**Comment by smjenness** [comment]
Similarly, see the first extension example in the JSS2 script (the Age-Dependent Mortality Extension model). Why do you specify the prevalence and network resimulation functions here?

```r
control <- control.net(type = NULL, nsims = 5, nsteps = 500,
                       departures.FUN = dfunc, arrivals.FUN = afunc,
                       prevalence.FUN = prevalence.net, infection.FUN = infection.net,
                       resim_nets.FUN = resim_nets, aging.FUN = aging)
```

**Comment by smjenness** [comment]
Also, with that first extension example, the EpiModel 1.x iteration in the JSS1 script used the `depend = TRUE` control setting to make sure that the network was resimulated. You do not have that specified correctly now with `resimulate.network` because the default value is FALSE.

**Comment by smjenness** [comment]
back to the default network resim and prevalence modules, here's the relevant text in the migration doc:

> Standard modules (in red) are typically not meant to be modified by the user in most applied extension cases. Those modules of class flexible (in blue) are often modified based on novel research cases and are typically related to disease modeling. For built-in models, the protected modules have hard-coded default functions in `control.net`, while the flexible modules have no hard-coded default functions because the appropriate function is selected based on the input parameters from `param.net` and initial conditions in `init.net`.

![Screen Shot 2020-07-08 at 6 31 21 PM](https://user-images.githubusercontent.com/2521570/86976791-3f7ab080-c149-11ea-99bb-33885fd1d7e5.png)

That is correct and what we agreed on for the design. But then here are the default inputs for control.net:

```r
control.net <- function(type,
                        nsteps,
                        start = 1,
                        nsims = 1,
                        ncores = 1,
                        resimulate.network = FALSE,
                        tergmLite = FALSE,
                        attr.rules,
                        epi.by,
                        initialize.FUN = initialize.net,
                        resim_nets.FUN = resim_nets,
                        infection.FUN = NULL,
                        recovery.FUN = NULL,
                        departures.FUN = NULL,
                        arrivals.FUN = NULL,
                        nwupdate.FUN = nwupdate.net,
                        prevalence.FUN = NULL,
                        verbose.FUN = verbose.net,
                        module.order = NULL,
                        set.control.ergm,
                        set.control.stergm,
                        save.nwstats = TRUE,
                        nwstats.formula = "formation",
                        save.other,
                        verbose = TRUE,
                        verbose.int = 1,
                        skip.check = FALSE,
                        raw.output = FALSE,
                        ...) {
```

**Comment by andsv2** [comment]
**prevalence.FUN** = NULL: the prevalence module is set to NULL in the control settings to allow it to be assigned based on user input. Right now the code is set up such that, crosscheck runs through the FUN modules and assigns in place any missing functions based on whether any of the functions are missing. Originally, prevalence.FUN was set up with the default, but because it's no longer NULL, it wasn't getting assigned its .2g version. So this was changed to be initially set to NULL. We can change this by putting in special handling for prevalence.FUN or, alternatively, consolidating the .2g workflow into the base prevalence.FUN module and setting a logic switch if it is not expected for users to  modify this module.

**SEIR Extension Model**: When type = NULL, we require the user to specify any NULL functions themselves; combined with the above this would require users looking to extend their modules to specify the prevalence function they wish to use as well.

**Comment by andsv2** [comment]
> Also, with that first extension example, the EpiModel 1.x iteration in the JSS1 script used the `depend = TRUE` control setting to make sure that the network was resimulated. You do not have that specified correctly now with `resimulate.network` because the default value is FALSE.

Updated.

**Comment by andsv2** [comment]
> Similarly, see the first extension example in the JSS2 script (the Age-Dependent Mortality Extension model). Why do you specify the prevalence and network resimulation functions here?
> 
> ```r
> control <- control.net(type = NULL, nsims = 5, nsteps = 500,
>                        departures.FUN = dfunc, arrivals.FUN = afunc,
>                        prevalence.FUN = prevalence.net, infection.FUN = infection.net,
>                        resim_nets.FUN = resim_nets, aging.FUN = aging)
> ```

Removed resim_net.FUN; prevalence.FUN explained below.

**Comment by andsv2** [comment]
> back to the default network resim and prevalence modules, here's the relevant text in the migration doc:
> 
> > Standard modules (in red) are typically not meant to be modified by the user in most applied extension cases. Those modules of class flexible (in blue) are often modified based on novel research cases and are typically related to disease modeling. For built-in models, the protected modules have hard-coded default functions in `control.net`, while the flexible modules have no hard-coded default functions because the appropriate function is selected based on the input parameters from `param.net` and initial conditions in `init.net`.
> 
> ![Screen Shot 2020-07-08 at 6 31 21 PM](https://user-images.githubusercontent.com/2521570/86976791-3f7ab080-c149-11ea-99bb-33885fd1d7e5.png)
> 
> That is correct and what we agreed on for the design. But then here are the default inputs for control.net:
> 
> ```r
> control.net <- function(type,
>                         nsteps,
>                         start = 1,
>                         nsims = 1,
>                         ncores = 1,
>                         resimulate.network = FALSE,
>                         tergmLite = FALSE,
>                         attr.rules,
>                         epi.by,
>                         initialize.FUN = initialize.net,
>                         resim_nets.FUN = resim_nets,
>                         infection.FUN = NULL,
>                         recovery.FUN = NULL,
>                         departures.FUN = NULL,
>                         arrivals.FUN = NULL,
>                         nwupdate.FUN = nwupdate.net,
>                         prevalence.FUN = NULL,
>                         verbose.FUN = verbose.net,
>                         module.order = NULL,
>                         set.control.ergm,
>                         set.control.stergm,
>                         save.nwstats = TRUE,
>                         nwstats.formula = "formation",
>                         save.other,
>                         verbose = TRUE,
>                         verbose.int = 1,
>                         skip.check = FALSE,
>                         raw.output = FALSE,
>                         ...) {
> ```

Reasoning explained above; can change language here to make this clearer and avoid confusion.

**Comment by andsv2** [comment]
> The `SIRwithVitalDynamics.R` file also does not run without error. It is missing an `r.num` initial condition.

Updated.

**Comment by smjenness** [comment]
@andsv2 : please confirm that you have run through all the files in inst/EpiModel2 and inst/JSS and they run without error.

**Comment by smjenness** [comment]
Fixed in #425
